From aeac2b531cb2c7f503330795753932862c9ea04a Mon Sep 17 00:00:00 2001
From: Praveen Diwakar <praveen.diwakar@intel.com>
Date: Mon, 6 Apr 2015 18:50:53 +0530
Subject: [PATCH] ALSA: compress: Exit from snd_compr_stop for
 SNDRV_PCM_STATE_OPEN

snd_compr_stop should not handle stream in open state i.e
SNDRV_PCM_STATE_OPEN.

Change-Id: Ia749cfd04fa44a35b24ef6589d22532aac41fb51
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-8131
Signed-off-by: Subhankar Paul <subhankar.paul@intel.com>
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
---
 sound/core/compress_offload.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/core/compress_offload.c b/sound/core/compress_offload.c
index c5ce629..d607af2 100644
--- a/sound/core/compress_offload.c
+++ b/sound/core/compress_offload.c
@@ -671,12 +671,12 @@ static int snd_compr_start(struct snd_compr_stream *stream)
 
 int snd_compr_stop(struct snd_compr_stream *stream)
 {
-	int retval;
+	int retval = 0;
 
-	if (stream->runtime->state == SNDRV_PCM_STATE_PREPARED ||
-			stream->runtime->state == SNDRV_PCM_STATE_SETUP)
+	if (stream->runtime->state == SNDRV_PCM_STATE_OPEN)
 		return -EPERM;
-	retval = stream->ops->trigger(stream, SNDRV_PCM_TRIGGER_STOP);
+	if (stream->runtime->state != SNDRV_PCM_STATE_PREPARED)
+		retval = stream->ops->trigger(stream, SNDRV_PCM_TRIGGER_STOP);
 	if (!retval) {
 		snd_compr_drain_notify(stream);
 		stream->runtime->total_bytes_available = 0;
-- 
1.9.1

